docs/protocol/architecture.mdx

```diff
@@ -0,0 +1,149 @@
+    end
+    
+    subgraph OnChain["On-Chain Layer (Smart Contracts)"]
+        CONTRACTS[Custody Contracts]
+    end
+    
+    subgraph Blockchain["Blockchain Layer"]
+        CHAIN[Ethereum, Polygon, etc.]
+    end
+    
+    APP --> CLIENT
+    APP --> BROKER
+    CLIENT <-->|Nitro RPC| BROKER
+    CLIENT --> CONTRACTS
+    BROKER --> CONTRACTS
+    CONTRACTS --> CHAIN
+    
+    style Application fill:#e1f5ff
+    style OffChain fill:#fff4e1
+    style OnChain fill:#ffe1f5
+    style Blockchain fill:#f0f0f0
+```
+
+## Communication Patterns
+
+### Channel Opening
+
+The <Tooltip content="A secure communication pathway between participants that locks funds in an on-chain smart contract while enabling off-chain state updates.">channel</Tooltip> opening process follows a coordinated sequence between client and <Tooltip content="A virtual ledger layer that provides a unified ledger (through Nitro RPC) and coordinates state channels (through Nitrolite), providing chain abstraction for developers and users.">a clearnode</Tooltip>:
+
+1. Client requests channel creation from <Tooltip content="A virtual ledger layer that provides a unified ledger (through Nitro RPC) and coordinates state channels (through Nitrolite), providing chain abstraction for developers and users.">a clearnode</Tooltip> via <Tooltip content="The off-chain communication protocol.">Nitro RPC</Tooltip>
+2. The clearnode returns signed initial <Tooltip content="A snapshot of the channel at a point in time, including fund allocations and application-specific data.">state</Tooltip>
+3. Client submits transaction to <Tooltip content="The underlying distributed ledger technology (e.g., Ethereum).">blockchain</Tooltip>
+4. The clearnode joins the channel on-chain
+5. Channel becomes active
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant Clearnode
+    participant Blockchain
+    
+    Client->>Clearnode: create_channel request
+    Clearnode->>Client: signed initial state
+    Client->>Blockchain: submit create() transaction
+    Blockchain->>Clearnode: Created event
+    Clearnode->>Blockchain: submit join() transaction
+    Blockchain->>Clearnode: Opened event
+    Clearnode->>Client: channel_update notification
+```
+
+:::success Cooperative Opening
+Channel opening requires cooperation between both parties, ensuring mutual agreement before funds are locked.
+:::
+
+### Off-Chain Updates
+
+Off-chain updates occur entirely between <Tooltip content="An entity (identified by a wallet address) that is part of a channel.">participants</Tooltip> without touching the blockchain:
+
+1. Participants exchange signed <Tooltip content="A snapshot of the channel at a point in time, including fund allocations and application-specific data.">state</Tooltip> updates via <Tooltip content="The off-chain communication protocol.">Nitro RPC</Tooltip>
+2. No blockchain transactions required
+3. Latest valid state maintained off-chain
+4. Can be checkpointed on-chain at any time
```

Comment:
In the ideal world - yes, but currently we don't store this off-chain state, because it's not related to the on-chain state, so we can't even checkpoint as well. But we are working on it, and I believe it will be more enforced in the next version of the protocol

Reply:
[2407a50f77b866e3cddc84621a441c69e6787515>](https://github.com/layer-3/docs/pull/69/commits/2407a50f77b866e3cddc84621a441c69e6787515)